### PR TITLE
Feature/new response handler

### PR DESCRIPTION
**Pull request by denselmann**

### CODE REVIEW

File: spec/support/mocks/states/prompt.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {
   PlatformGenerator,
   State,
 } from "assistant-source";
+// tslint:disable-next-line:no-implicit-dependencies
```
Comment (Novaal) [inline spec/support/mocks/states/prompt.ts]: This line should not be necessary

File: spec/support/mocks/states/prompt.ts [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
+  BaseState,
+  BasicAnswerTypes,
+  BasicHandable,
+  CurrentSessionFactory,
+  EntityDictionary,
+  injectionNames,
+  PlatformGenerator,
+  State,
+} from "assistant-source";
+// tslint:disable-next-line:no-implicit-dependencies
```
Comment (antoniusostermann) [inline spec/support/mocks/states/prompt.ts]: You shouldn't need this

File: spec/support/call-intent-helper.ts [hunk 3]
```diff
@@ -1,39 +1,46 @@
-import { PlatformSpecHelper, intent } from "assistant-source";
+// tslint:disable-next-line:no-implicit-dependencies
```
Comment (Novaal) [inline spec/support/call-intent-helper.ts]: This should not be necessary. As this dependency is only used in the specs you can add a tslint.json in the specs folder with the following content:

```json
{
  "extends": "../tslint.json",
  "rules": {
    "no-implicit-dependencies": false
  }
}
```

File: spec/support/mocks/states/prompt.ts [hunk 4]
```diff
@@ -1,14 +1,25 @@
...
+import { inject, injectable } from "inversify";
 
-/** 
- * This small class is needed to apply the PromptStateMixin since TypeScript does not allow type-specific constructor mixins. 
+/**
+ * This small class is needed to apply the PromptStateMixin since TypeScript does not allow type-specific constructor mixins.
  * Just add it to your regular class hierarchy.
  */
-class PromptStateRequirements extends BaseState implements PromptStateMixinRequirements {
+class PromptStateRequirements<CustomTypes extends BasicAnswerTypes, CustomHandler extends BasicHandable<CustomTypes>>
```
Comment (Novaal) [inline spec/support/mocks/states/prompt.ts]: I would use the word `MergedHandler` and `MergedAnswerTypes` here, to fit to the wording used in AssistantJS

File: spec/support/mocks/states/prompt.ts [hunk 5]
```diff
@@ -17,14 +28,17 @@ class PromptStateRequirements extends BaseState implements PromptStateMixinRequi
   }
 }
 
+// tslint:disable-next-line:max-classes-per-file
 @injectable()
-export class PromptState extends PromptStateMixin(PromptStateRequirements) {
+export class PromptState<CustomTypes extends BasicAnswerTypes, CustomHandler extends BasicHandable<CustomTypes>> extends PromptStateMixin(
```
Comment (Novaal) [inline spec/support/mocks/states/prompt.ts]: Rename here again, see above.

File: src/components/authentication/strategies/pin.ts [hunk 6]
```diff
@@ -1,9 +1,10 @@
-import { injectable, inject, optional } from "inversify";
-import { injectionNames, ComponentSpecificLoggerFactory, Session, EntityDictionary, Transitionable, Logger } from "assistant-source";
+import { ComponentSpecificLoggerFactory, EntityDictionary, injectionNames, Logger, Session, Transitionable } from "assistant-source";
+// tslint:disable-next-line:no-implicit-dependencies
```
Comment (Novaal) [inline src/components/authentication/strategies/pin.ts]: This should not be necessary. Add `assistant-validations` to Dev- and PeerDependencies

**Comment by Novaal** [comment]
Please merge https://github.com/Novaal/assistant-authentication/tree/feature/new-response-handler into your Branch and everything should be ok.

**Pull request merged**